tests/pass-dep/libc/libc-mem.rs

```diff
@@ -190,11 +190,10 @@ fn test_memalign() {
         let align = 64;
         let size = 0;
         assert_eq!(libc::posix_memalign(&mut ptr, align, size), 0);
-        // We are not required to return null if size == 0, but we currently do.
-        // It's fine to remove this assert if we start returning non-null pointers.
-        assert!(ptr.is_null());
+        // Non-null pointer is returned if size == 0
+        assert!(!ptr.is_null());
+        //TODO: The line below has caused a leak?
         assert!(ptr.is_aligned_to(align));
-        // Regardless of what we return, it must be `free`able.
         libc::free(ptr);
```

Comment:
The test will pass without error when line 196 is commented out, and memory leak is reported when it is included. But I don't think a memory leak should be reported since ``ptr`` is freed afterwards? 

Command used: ``./miri run --dep ./tests/pass-dep/libc/libc-mem.rs``

Reply:
Resolved: Align::ONE shouldn't be used here.